students/WooseokJ/TrigramWJ.py

```diff
@@ -0,0 +1,37 @@
+import sys 
+import random
+
+#takes the words from the given file and returns it in a list
+def words_from_file(filename):
+    with open(filename, 'r') as textfile: 
+        return [word.strip('\n') for l in textfile.readlines() 
+                for word in l.split(' ')] 
+#from the list makes a dictiondary to mix it in the next process
+def build_trigrams_dict(words):
+    trigrams = {}
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+        if pair not in trigrams:
+            trigrams[pair] = [follower]
+        else:
+            trigrams[pair] += [follower]
+    return trigrams
+#this mixes the words to create the new text
+def build_text(pairs):
+    random_text = []
+    while len(random_text) < len(pairs):
+        random_key = random.choice(list(pairs.keys()))
+        value = pairs.get(random_key)
+        random_text.append(list(random_key))
+        random_text.append(value)
+        new_list = [item for fixedlist in random_text for item in fixedlist]
+        clean_text = " ".join(new_list)
+#this is a promot that asks for the file that we want to scramble make sure to add .txt at the end of the file
+        return clean_text
```

Comment:
Your code is working correctly up until _build_trigrams_dict()_. You only need to select a random key from the trigrams dictionary once, after that, you need to to continue the sequence. It should work like this:

1. Select a random key from _pairs_ (which is really the trigrams dictionary).
2. Add the two words in that random key to _random_text_.
3. Get the list of words associated to _random_key_ from the pairs dictionary.
4. Use random again to select one word from that list of words, let's call it _selected_word_.
5. Create the next key by taking the second word in the current key and _selected_word_.
6. Get the list of words associated with the new key.
7. Select a random word from that list.
8. Add the word to _random_text_.
9. Repeat steps 5 to 8 until you find a key that is not in the dictionary and/or until your _random_text_ reaches a certain number of words.